src/main/java/org/hmhb/authentication/DefaultJwtAuthenticationService.java

```diff
@@ -0,0 +1,178 @@
+
+        String authHeader = request.getHeader("Authorization");
+
+        LOGGER.debug("processing authentication: header={}", authHeader);
+
+        if (authHeader != null) {
+
+            String[] authHeaderInfo = authHeader.split("[ ]");
+
+            if (authHeaderInfo.length < 2) {
+                throw new AuthHeaderMissingTokenException(authHeader);
+            }
+
+            if (authHeaderInfo.length > 2) {
+                throw new AuthHeaderHasTooManyPartsException(authHeader);
+            }
+
+            if (!"Bearer".equals(authHeaderInfo[0])) {
+                throw new AuthHeaderUnknownException(authHeader);
+            }
+
+            String authToken = authHeaderInfo[1];
+
+            if (!Jwts.parser().isSigned(authToken)) {
+                throw new JwtTokenNotSignedException();
+            }
+
+            String domainId = getDomain();
+            String secret = getSecret();
+
+            byte[] keyBytes = DatatypeConverter.parseBase64Binary(secret);
+            /* I'm using setSigningKey with the SecretKeySpec so I can enforce which algorithm to use. */
+            SecretKeySpec secretKey = new SecretKeySpec(keyBytes, SignatureAlgorithm.HS256.getJcaName());
+
+            LOGGER.debug("authHeader is: {}", authHeader);
+            LOGGER.debug("authToken is: {}", authToken);
+
+            Claims claims = Jwts.parser()
+                    .requireAudience(domainId)
+                    .requireIssuer(domainId)
+                    .setSigningKey(secretKey)
+                    .parseClaimsJws(authToken)
+                    .getBody();
+
+            String username = claims.getSubject();
+            Date issuedAt = claims.getIssuedAt();
+            Date expirationDate = claims.getExpiration();
+            String issuer = claims.getIssuer();
+            String audience = claims.getAudience();
+            boolean admin = claims.get("admin", Boolean.class);
+            String email = claims.get("email", String.class);
+            Long userId = claims.get("userId", Integer.class).longValue();
+
+            Date now = new Date();
+
+            if (issuedAt.after(now)) {
+                throw new JwtTokenNotActiveYetException(issuedAt);
+            }
+
+            if (expirationDate.before(now)) {
+                throw new JwtTokenExpiredException(expirationDate);
+            }
+
+            LOGGER.debug(
+                    "token accepted, standard info: issuer={}, audience={}, username={}, issuedAt={}, expiresOn={}",
+                    issuer,
+                    audience,
+                    username,
+                    issuedAt,
+                    expirationDate
+            );
+
+            LOGGER.debug(
+                    "token accepted, custom info: admin={}, email={}, userId={}",
+                    admin,
+                    email,
+                    userId
+            );
+
+            HmhbUser user = new HmhbUser();
```

Comment:
Behold the magic of JWT.  I don't have to look up the user's info in the DB again.  I encoded that info in the generateJwtToken when they first logged in and here in validateAuthentication I can extract that info and trust it because I signed it w/ our secret in the generateJwtToken.